### PR TITLE
Fix: Migrate table schemas only for snapshots with the FORWARD_ONLY category explicitly assigned

### PR DESCRIPTION
**Pull request by izeigerman**

### CODE REVIEW

**Review by tobymao — APPROVED** [review]

File: sqlmesh/core/console.py [hunk 1]
```diff
@@ -29,9 +29,13 @@
 
 
 SNAPSHOT_CHANGE_CATEGORY_STR = {
+    None: "Unknown",
     SnapshotChangeCategory.BREAKING: "Breaking",
     SnapshotChangeCategory.NON_BREAKING: "Non-breaking",
     SnapshotChangeCategory.FORWARD_ONLY: "Forward-only",
+    SnapshotChangeCategory.INDIRECT_BREAKING: "Indirect breaking",
+    SnapshotChangeCategory.INDIRECT_NON_BREAKING: "Indirect non-breaking",
```
Comment (tobymao) [inline sqlmesh/core/console.py:37]: what's the difference between indirect non breaking and indirect forward only

File: sqlmesh/core/context_diff.py [hunk 2]
```diff
@@ -135,11 +140,13 @@ def create(
                 local_head = snapshot.previous_version.version
 
                 if remote_head in (local.version for local in snapshot.previous_versions):
-                    snapshot.set_version(local_head)
+                    snapshot.version = local_head
+                    snapshot.change_category = snapshot.previous_version.change_category
                 elif local_head in (remote.version for remote in remote_versions):
-                    snapshot.set_version(remote_head)
+                    snapshot.version = remote_head
+                    snapshot.change_category = remote_versions[-1].change_category
```
Comment (tobymao) [inline sqlmesh/core/context_diff.py:147]: why not remote_head.change_category?

File: sqlmesh/core/plan/definition.py [hunk 3]
```diff
@@ -292,22 +292,15 @@ def set_choice(self, snapshot: Snapshot, choice: SnapshotChangeCategory) -> None
...
 
         for child in self.indirectly_modified[snapshot.name]:
             child_snapshot = self.context_diff.snapshots[child]
 
             if choice == SnapshotChangeCategory.BREAKING:
-                child_snapshot.set_version()
+                child_snapshot.categorize_as(SnapshotChangeCategory.INDIRECT_BREAKING)
             else:
-                child_snapshot.set_version(child_snapshot.previous_version)
+                child_snapshot.categorize_as(SnapshotChangeCategory.INDIRECT_FORWARD_ONLY)
```
Comment (tobymao) [inline sqlmesh/core/plan/definition.py:303]: i noticed indirect-non-breaking is not used anywhere

File: sqlmesh/core/snapshot/definition.py [hunk 4]
```diff
@@ -42,12 +42,17 @@ class SnapshotChangeCategory(IntEnum):
...
+    FORWARD_ONLY: The change requires no rebuilding
+    INDIRECT_BREAKING: The change was caused indirectly and is breaking.
+    INDIRECT_FORWARD_ONLY: The change was caused indirectly and is forward-only.
     """
 
     BREAKING = 1
     NON_BREAKING = 2
     FORWARD_ONLY = 3
+    INDIRECT_BREAKING = 4
+    INDIRECT_NON_BREAKING = 5
```
Comment (tobymao) [inline sqlmesh/core/snapshot/definition.py:54]: i guess delete this?

File: sqlmesh/core/console.py (hunk 1 above)
Comment (izeigerman) [reply sqlmesh/core/console.py:37]: Indirect non-breaking is reserved for future use in which the snapshot itself needs to be rebuilt but its further dependencies - not.

File: sqlmesh/core/plan/definition.py (hunk 3 above)
Comment (izeigerman) [reply sqlmesh/core/plan/definition.py:306]: See comment: https://github.com/TobikoData/sqlmesh/pull/763#discussion_r1177086530

**Comment by tobymao** [comment]
looks good

**Pull request merged**